target/linux/lantiq/dts/RS353.dts

```diff
@@ -0,0 +1,318 @@
+/dts-v1/;
+
+/include/ "vr9.dtsi"
+
+/ {
+	model = "RS353 - Bintec Elmeg RS353";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200 init=/etc/preinit";
+		leds {
+                        running = &status;
+                        boot = &status;
+			wifi = &wlan;
+			usb2 = &usb; /* Front port is mapped to device 2 */
+			dsl = &dsl;
+                };
+
+	};
+
+	memory@0 {
+		reg = <0x0 0x8000000>;
+	};
+
+	sram@1F000000 {
+		cgu0: cgu@103000 {
+                        compatible = "lantiq,cgu-xway";
+                        reg = <0x103000 0x1000>;
+                        #clock-cells = <1>;
+			lantiq,phy-clk-src = <0x2>;
+		};
+	};
+
+	fpi@10000000 {
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x2000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					boardconfig: partition@0 {
```

Comment:
Okay, the bootloader layout is really strange. What I can see so far:

mtd0 (0x0): is some kind of preloader, which loads the next step from mtd1. beside from that, the partition is more or less empty.

mtd1 (0x20000): seams to be the real bootloader and is called Logic

mtd2 (0x40000): yet another bootloader which is called Bootmonitor

I've no idea what is the purpose of the individual bootloaders and why they have chosen such a complex boot layout. But I guess it has some special reason.

Anyway, I would recommend the following naming:

mtd0 - preloader
mtd1 - logic
mtd2 - bootmonitor

But I guess I saw the reason for your extra uboot + mac address hack. The whole bintec system seams to run in big endian mode but OpenWrt lantiq support is little endian. Am I right that the purpose of your extra u-boot is to switch the SoC back to little endian?